tools/README.md

```diff
@@ -32,3 +40,93 @@ Contents of the config file must be in the following format:
 `txAsync` defines, whether `/tx` or `/tx/async` endpoint is used.
 
 Configuration file suitable for `./scripts/launch/node w` is already present in this directory for convenience.
+
+## keygen
+
+This is a small tool to manipulate with secret key. Along with generating new one, it allows to produce various deriviative data.
+
+### Usage
+
+Main usage pattern is
+```bash
+echo "<commands>" | dscp-keygen <base-secret-option>
```

Comment:
This is nice, but why not use `-c|--command` option instead of reading from stdin? It would be much clearer, and those who would like to supply command from input stream for some reason (e. g. redirecting from some script) could use `xargs` anyway.

If anything is worth reading from stdin, it's the _input_ of the command - which could be a seed, secret or a keyfile. Then the usage of the keygen would be something very similar to `sed`, like:
```
cat ./secret.key | dscp-keygen --keyfile --command "address"   # display the address
```
or 
```
echo "abcde123" | dscp-keygen --seed --command "keyfile:qwerty123:pretty"
```
Even `--random` option might be omitted, because you can just do
```
take -c 10 /dev/urandom | dscp-keygen --seed --command "secret:base64"
```
I think such approach is more straightforward and Unix-like.

Reply:
Right, this makes perfect sense.